lms/djangoapps/commerce/api/v1/views.py

```diff
@@ -64,10 +73,17 @@ class OrderView(APIView):
     """ Retrieve order details. """
 
     authentication_classes = (SessionAuthentication,)
-    permission_classes = (IsAuthenticated,)
+    permission_classes = (IsAuthenticatedOrActivationOverridden,)
```

Comment:
The IsAuthenticated permission class will return False if it's an AnonymousUser, which is what I was seeing for the request.user value when my test user was authenticated/logged in.  I'm not really sure why that was happening and assumed it was because the user was not active and something in the authentication middleware is recognizing that and setting request.user to AnonymousUser.